src/gui/qgsattributeformeditorwidget.cpp

```diff
@@ -214,7 +215,19 @@ void QgsAttributeFormEditorWidget::updateWidgets()
 {
   //first update the tool buttons
   bool hasMultiEditButton = ( editPage()->layout()->indexOf( mMultiEditButton ) >= 0 );
-  bool fieldReadOnly = layer()->editFormConfig().readOnly( mWidget->fieldIdx() );
+
+  bool fieldReadOnly = false;
```

Comment:
This looks almost identical to QgsAttributeForm::fieldIsEditable. Could we make a common function for this instead to avoid the extra code?

Reply:
Indeed, and [here](https://github.com/qgis/QGIS/blob/master/src/gui/attributetable/qgsattributetablemodel.cpp#L767) also.

I added a method in QgsVectorLayerUtils. Hope it's the right place